src/dbnode/storage/series/reader.go

```diff
@@ -203,6 +203,52 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 	return results, nil
 }
 
+// FetchIndexChecksum reads index checksum blocks using just a block retriever.
+func (r Reader) FetchIndexChecksum(
+	ctx context.Context,
+	blockStart time.Time,
+	nsCtx namespace.Context,
+) (ident.IndexChecksum, error) {
+	return r.fetchIndexChecksum(ctx, blockStart, nsCtx)
+}
+
+func (r Reader) fetchIndexChecksum(
+	ctx context.Context,
+	blockStart time.Time,
+	nsCtx namespace.Context,
+) (ident.IndexChecksum, error) {
+	var (
+		nowFn        = r.opts.ClockOptions().NowFn()
+		now          = nowFn()
+		ropts        = r.opts.RetentionOptions()
+		size         = ropts.BlockSize()
+		alignedStart = blockStart.Truncate(size)
+	)
+	// Squeeze the lookup window by what's available to make range queries like [0, infinity) possible
+	earliest := retention.FlushTimeStart(ropts, now)
+	if alignedStart.Before(earliest) {
+		alignedStart = earliest
+	}
```

Comment:
Hm don't think this is correct (squeezing it to the last block).

We only do that in the other method since we're doing a scan of all known blocks. This is for an exact block start. If it's looking for block start out of retention we should return empty/not found instead of a potential result from the last block.

Reply:
Good call :+1: